src/utils/helpers/focus-trap/focus-trap.js

```diff
@@ -5,48 +5,55 @@ function blockTabbing(ev) {
   }
 }
 
-function trap(firstFocusableElement, lastFocusableElement, ev) {
+function setElementFocus(element) {
+  if (typeof element === 'function') {
+    element();
+  } else {
+    element.focus();
+  }
+}
+
+function trap(firstFocusableElement, lastFocusableElement, bespokeTrap, ev) {
+  if (bespokeTrap) {
+    bespokeTrap(ev, firstFocusableElement, lastFocusableElement);
+    return;
+  }
+
   if (ev.key === 'Tab' || ev.keyCode === 9) {
     if (ev.shiftKey) /* shift + tab */ {
-      if (typeof firstFocusableElement === 'function') {
-        ev.preventDefault();
-        if (document.activeElement !== lastFocusableElement) {
-          lastFocusableElement.focus();
-        } else {
-          firstFocusableElement();
-        }
-      } else if (document.activeElement === firstFocusableElement) {
+      if (document.activeElement === firstFocusableElement) {
         lastFocusableElement.focus();
         ev.preventDefault();
       }
     } else if (document.activeElement === lastFocusableElement) {
-      if (typeof firstFocusableElement === 'function') {
-        firstFocusableElement();
-      } else {
-        firstFocusableElement.focus();
-      }
+      setElementFocus(firstFocusableElement);
       ev.preventDefault();
     }
   }
 }
+
 // eslint-disable-next-line max-len
 const defaultFocusableSelectors = 'button, [href], input:not([type="hidden"]), select, textarea, [tabindex]:not([tabindex="-1"])';
 
-const focusTrap = (element, focusFirstElement) => {
+const focusTrap = (element, focusFirstElement, bespokeTrap) => {
```

Comment:
I noticed that `focusFirstElement` can have three values, `true`, `false` and `function(){}` I think we should make this two separate arguments for clarity.

```suggestion
const focusTrap = (element, autoFocus = true, focusFirstElement, bespokeTrap) => {
```
`focusFirstElement` can be a function or a DOMElement.

Then this

```js
if (focusFirstElement) {
    if (typeof focusFirstElement === 'boolean') {
      setElementFocus(firstFocusableElement);
    } else {
      setElementFocus(focusFirstElement);
    }
  }
```
would become
```
if (autoFocus) {
     setElementFocus(focusFirstElement || focusableElements[0]);
}
```